### DISCUSSION TITLE
Validating Array and display errors

### DISCUSSION THREAD

**Discussion by mnlfischer** [react-hook-form/react-hook-form#3800] [Q&A] [answered]
Is there a better solution to check if the returned error object is of type array in case of having a required array and required props of array items?
Check example:
https://codesandbox.io/s/react-hook-form-usefieldarray-forked-ih4vk?file=/src/index.js

In case of missing array items, i check if errors exists and check its type:

```
{errors.persons && !Array.isArray(errors.persons) && (
  <span style={{ color: "red" }}>Persons are required</span>
)}
```

The current error is an object:
```
persons: Object
  message: "persons is a required field"
  type: "required"
```

When there are items in the array but errors still exists based on its validation, is can display them via:
```
  {errors.persons?.[index]?.lastName && (
    <span style={{ color: "red" }}>LastName is required</span>
  )}
```

The current error is an array of objects:
```
persons: Array(1)
  0: Object
    firstName: Object
    message: "persons[0].firstName is a required field"
    type: "required"
    lastName: Object
    message: "persons[0].lastName is a required field"
    type: "required"

```

The array required error is still present if an item was added, even if I use trigger to revalidate. Is that a known behavior? I think it would be nice if the error is removed like the behavior on a normal input element when it is touched.

**Comment by bluebill1049** [comment, marked as answer]
Is this related min length of the array field?

**Reply by mnlfischer** [reply]
I found the fieldArrayMinLength.tsx example and solved it with clearing the error state and solve type errors with the NestedValue type. Thanks for your great work!

**Reply by bluebill1049** [reply]
we are thinking of a way to improve this, the tricky part is errors object for the field array is an array, while with minLength, maxLength we sort need an object. a case like the following make it even harder:

`minLength` combined with some fields are `required`.

drop us a comment if you have any good idea as well: https://github.com/react-hook-form/react-hook-form/discussions/3714 we can make v7 even greater.
